content/docs/buildpack-basics.md

```diff
@@ -0,0 +1,63 @@
++++
+title="Basics of BuildPacks"
+weight=2
+getting-started=true
++++
+
+## Basic Concepts
+
+### What is a Buildpack?
+
+A `buildpack` is software that transforms application source code into
+executable files by analyzing the code and determining the best way to
+build it.
+
+![buildpacks](/images/what.svg)
+
+### What is a Builder?
+
+A builder is an image that contains all the components necessary to
+execute a build (for example, an ordered combination of buildpacks, a build
+image and other files and configurations).
+
+![create-builder diagram](/images/create-builder.svg)
+
+### What is a Lifecycle?
+
+A lifecycle is a series of steps that are used to create and manage a
+buildpack. `create` is used to analyze, detect, restore, build, and export
+buildpack execution. Next, `launcher` can be used to launch the application.
+Finally, `rebase` can be used to push the latest changes to an existing
+buildpack. All of these steps are part of a lifecycle.
```

Comment:
Lifecycle and launcher can be used independently of each other, should we mix them both in the same initial description?

Reply:
- How about the following (I guess the image may also need to be updated, please let me know the required changes):

_A lifecycle is a series of steps that are used to create and manage a buildpack. The cumulative `create` step can be used to `analyze`, `detect`, `restore`, `build`, and `export` buildpack execution. All of these steps are part of a lifecycle. You can also re-enter a lifecycle using `rebase` to push the latest changes to an existing buildpack._

~[image]()

_`launcher` is an independent step that can be used to launch the application at any time._